src/conversions/osstr.rs

```diff
@@ -58,7 +58,22 @@ impl ToPyObject for OsStr {
 
 impl FromPyObject<'_> for OsString {
     fn extract(ob: &PyAny) -> PyResult<Self> {
-        let pystring = <PyString as PyTryFrom>::try_from(ob)?; // Cast PyAny to PyString
+        let pystring = match <PyString as PyTryFrom>::try_from(ob) {
+            // Cast PyAny to PyString
+            Ok(s) => s,
+            Err(err) => {
+                let py = ob.py();
+                let pathlib = py.import("pathlib")?;
+                let pathlib_path = pathlib.getattr("Path")?;
+                // FIXME: right way to get the type object of pathlib.Path?
+                let path_instance = pathlib_path.call0()?;
+                if path_instance.get_type().is_instance(ob)? {
```

Comment:
You already got the type object just above ;)

```suggestion
                let pathlib_path: &PyType = pathlib.getattr("Path")?.downcast()?;
                if pathlib_path.is_instance(ob)? {
```

Reply:
😅 Ahh, it's not very clear to me that you can get `&PyType` by downcasting.